### DISCUSSION TITLE
目前未把DDL和DML分开，导致Oracle出现ORA-00904

### DISCUSSION THREAD

**Discussion by planet-zjx** [hhyo/Archery#2065] [General]
把添加A字段和更新A字段放在一个工单里面，因为有依赖关系，导致出现ORA-00904报错，这个除了DDL和DML分开提工单，是否还有其他解决办法。

**Comment by javaanddonet** [comment]
你可以提交create table的DDL语句吗？我在流程里提交不了。不支持create table的DDL，但是alter table的DDL却支持。

**Comment by planet-zjx** [comment]
看到有这个参数可以控制一个工单里面不能同时共存DDL和DML，但是设置时不认识这个参数er_ddl_dml_coexist = 2，详见https://github.com/hanchuanchuan/goInception/blob/master/config/config.toml.default，希望大佬解答下

**Reply by LeoQuote** [reply]
goinception 不可能负责 oracle 相关的逻辑, 你说的可能和本问题不相关

**Reply by planet-zjx** [reply]
那有什么办法在工单提交时“同时存在DDL+DML”可以不允许提交工单呢

**Reply by LeoQuote** [reply]
当前 oracle 引擎应该不支持解析语句, 或者只有简单解析, 你说的这个功能应该更没有了, 欢迎pr

**Reply by planet-zjx** [reply]
那针对MySQL有什么办法强制分离DDL和DML，就是两者不能同时出现在一个工单里面，就像这个参数er_ddl_dml_coexist=2，同时出现DDM+DML就会报错，不允许提交。
